Lib/wave.py

```diff
@@ -495,6 +499,13 @@ def open(f, mode=None):
             mode = f.mode
         else:
             mode = 'rb'
+    try:
+        f = os.fspath(f)
+    except TypeError:
+        if not hasattr(f, 'read') or not hasattr(f, 'close'):
```

Comment:
I'd also check for a `write` method here.

Reply:
I think it is enough to check for a `read` in `Wave_read` and for a `write` on `Wave_write`. `close` should be optional.

Note that in `Wave_read` and `Wave_write` constructors there are checks `isinstance(f, str)`. This means that bytes patch are not supported, and file descriptors are not supported. I would use `hasattr(f, 'read')` or `hasattr(f, 'write')` instead and just pass other objects to `builtins.open()`.